dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/webapp/WebApp.java

```diff
@@ -1319,21 +1315,22 @@ public ManagedObject inject(final Object target) throws InjectionException
                
                 try {
                     ModuleMetaData moduleMetaData = getModuleMetaData();
-                    mo = managedObjectService.createManagedObject(moduleMetaData, target);
-                } catch (IllegalStateException ise){
+                    ManagedObjectFactory<Object> mof = (ManagedObjectFactory<Object>) managedObjectService.createManagedObjectFactory(moduleMetaData, target.getClass(), false);
+                    mo = mof.createManagedObject(target, null);
+                } catch (ManagedObjectException moe){
                     
                     if (TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
-                        logger.logp(Level.FINE, CLASS_NAME, METHOD_NAME, "IllegalStateException from managed object service:" + ise );
+                        logger.logp(Level.FINE, CLASS_NAME, METHOD_NAME, "IllegalStateException from managed object service:" + moe );
                      }   
                     
                      mo = new WCManagedObjectImpl(target);
                  }
               
                 r = mo;
-                injectTargets(mo);
+                mo.inject(this.referenceContext);
            }
-        } catch (ManagedObjectException e) {
-            throw new InjectionException(e.getCause());
+        } catch (Exception e) {
```

Comment:
is this exception handling right? the old one looks better somehow?

Reply:
The old one converts ManagedObjectExceptions into InjectionExceptions where as the new one wraps any exception inside an InjectionException. 

I prefer the new one as in my PMR experience you can never really have too much details in your log. But I'm not sure if it's "better". I'm leaving this as is for now.